components/camel-redis/src/main/java/org/apache/camel/component/redis/processor/aggregate/RedisAggregationRepository.java

```diff
@@ -0,0 +1,338 @@
+		this.endpoint = endpoint;
+	}
+
+	public RedisAggregationRepository(final String mapName, final String persistenceMapName,
+			final String endpoint) {
+		this.mapName = mapName;
+		this.persistenceMapName = persistenceMapName;
+		this.optimistic = false;
+		this.endpoint = endpoint;
+	}
+
+	public RedisAggregationRepository(final String mapName, final String endpoint, boolean optimistic) {
+		this(mapName, endpoint);
+		this.optimistic = optimistic;
+	}
+
+	public RedisAggregationRepository(final String mapName, final String persistenceMapName,
+			final String endpoint, boolean optimistic) {
+		this(mapName, persistenceMapName, endpoint);
+		this.optimistic = optimistic;
+	}
+
+	@Override
+    public Exchange add(CamelContext camelContext, String key, Exchange oldExchange, Exchange newExchange) throws OptimisticLockingException {
+        if (!optimistic) {
+            throw new UnsupportedOperationException();
+        }
+        LOG.trace("Adding an Exchange with ID {} for key {} in an optimistic manner.", newExchange.getExchangeId(), key);
+        if (oldExchange == null) {
+            DefaultExchangeHolder holder = DefaultExchangeHolder.marshal(newExchange, true, allowSerializedHeaders);
+            final DefaultExchangeHolder misbehaviorHolder = cache.putIfAbsent(key, holder);
+            if (misbehaviorHolder != null) {
+                Exchange misbehaviorEx = unmarshallExchange(camelContext, misbehaviorHolder);
+                LOG.error("Optimistic locking failed for exchange with key {}: IMap#putIfAbsend returned Exchange with ID {}, while it's expected no exchanges to be returned",
+                        key, misbehaviorEx != null ? misbehaviorEx.getExchangeId() : "<null>");
+                throw  new OptimisticLockingException();
+            }
+        } else {
+            DefaultExchangeHolder oldHolder = DefaultExchangeHolder.marshal(oldExchange, true, allowSerializedHeaders);
+            DefaultExchangeHolder newHolder = DefaultExchangeHolder.marshal(newExchange, true, allowSerializedHeaders);
+            if (!cache.replace(key, oldHolder, newHolder)) {
+                LOG.error("Optimistic locking failed for exchange with key {}: IMap#replace returned no Exchanges, while it's expected to replace one",
+                        key);
+                throw new OptimisticLockingException();
+            }
+        }
+        LOG.trace("Added an Exchange with ID {} for key {} in optimistic manner.", newExchange.getExchangeId(), key);
+        return oldExchange;
+    }
+
+	@Override
+	public Exchange add(CamelContext camelContext, String key, Exchange exchange) {
+		if (optimistic) {
+			throw new UnsupportedOperationException();
+		}
+		LOG.trace("Adding an Exchange with ID {} for key {} in a thread-safe manner.", exchange.getExchangeId(), key);
+		Config config = new Config();
+		config.useSingleServer().setAddress(String.format("redis://%s",endpoint));
+		RedissonClient redisson = Redisson.create(config);
```

Comment:
The global `redisson` instance is created by `onStart` so I guess this code can be replaced by using the shared instance